rest-client/src/main/java/org/sagebionetworks/bridge/sdk/rest/ApiClientProvider.java

```diff
@@ -0,0 +1,146 @@
+package org.sagebionetworks.bridge.sdk.rest;
+
+import com.fatboyindustrial.gsonjodatime.Converters;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+import org.sagebionetworks.bridge.sdk.rest.api.AuthenticationApi;
+import org.sagebionetworks.bridge.sdk.rest.model.SignIn;
+
+import java.lang.ref.WeakReference;
+import java.lang.reflect.Type;
+import java.util.Base64;
+import java.util.Map;
+
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * Created by liujoshua on 10/11/16.
+ */
+public class ApiClientProvider {
+  public static final Gson GSON = Converters
+            .registerAll(new GsonBuilder())
+            .registerTypeAdapter(byte[].class, new ByteArrayToBase64TypeAdapter())
+            .create();
+
+  private static class ByteArrayToBase64TypeAdapter implements JsonSerializer<byte[]>, JsonDeserializer<byte[]> {
+    public byte[] deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+      return Base64.getDecoder().decode(json.getAsString());
```

Comment:
We use four space for indentation across all other projects; I don't want to have to switch to 2 just for this project. I think this should follow the common formatting standard here.